src/character.cpp

```diff
@@ -2351,14 +2351,16 @@ void Character::practice( const skill_id &id, int amount, int cap, bool suppress
                      skill_name );
         }
 
-        int chance_to_drop = focus_pool;
-        focus_pool -= chance_to_drop / 100;
         // Apex Predators don't think about much other than killing.
         // They don't lose Focus when practicing combat skills.
-        if( ( rng( 1, 100 ) <= ( chance_to_drop % 100 ) ) && ( !( has_trait_flag( "PRED4" ) &&
-                skill.is_combat_skill() ) ) ) {
-            focus_pool--;
+        if( !( has_trait_flag( "PRED4" ) && skill.is_combat_skill() ) ) {
+            // Reduce by 1% of total, or practice amount * 10, whichever is larger.
+            // The latter kicks in when long actions like crafting
+            // apply many turns of gains at once.
+            int focus_drain = std::max( focus_pool / 100, amount );
+            focus_pool -= ( focus_drain * focus_drain ) / 1000;
```

Comment:
A couple of things:
- Line 2357 says practice amount*10, but 2360 seems not to have the *10.
- Line 2361 is giving an overflow error on some builds. Possibly have focus_drain be a double and do the division by 1000 and squaring as a double, then (possibly with random remainder handling) convert to an int for subtracting from focus_pool?